packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -581,22 +581,36 @@ call something like:
 driver.executeScript('soundz: playSong', [{song: 'Stairway to Heaven', artist: 'Led Zeppelin'}]);
 ```
 
-Then in your driver code you could simply look for `soundz:` at the beginning of your
-`execute` override, and do the appropriate thing:
+Then in your driver code you could simply define the magic static field `executeOverloadMap` as
+a mapping of script names to member functions on your driver. It has the same basic form as
+`newMethodMap`. All you need to do at that point is implement the `execute` the command to take
+advantage of the built-in `executeOverload` method available on all drivers:
 
 ```js
-async execute(script, args) {
-  if (script === 'soundz: playSong') {
-    const {song, artist} = args[0];
-    // play the song based on song and artist details
-    return;
-  }
+static executeOverloadMap = {
+  command: 'soundz: playSong',
+  parameters: {required: ['song', 'artist'], optional: []},
```

Comment:
This example code is wrong (`executeOverloadMap` is of the wrong shape)

Reply:
I'll buy you a sandwich if you change the API to accept a JSON schema instead of `{required, optional}`.  I can hook up the validation later.  I'd rather just not keep adding more of these DSLs for describing the shape of parameters

Reply:
that'd look like, for the above example (using the presumably correct object shape):

```js
static executeOverloadMap = {
  'soundz: playSong': {
    properties: {
      song: {type: 'string'},
      artist: {type: 'string'}
    },
    required: ['song', 'artist']
  }
}
```

We may be able to add a shorthand syntax, like:

```js
static executeOverloadMap = {
  'soundz: playSong': {
    'appium:stringProps': ['song', 'artist'],
    required: ['song', 'artist']
  }
}

```

either way, I think using JSON schema semantics here, while more verbose than what you originally had, is negligible when considering the utility the new API offers.

Reply:
i think the main reason not to do that now is it is the way we have defined all appium commands. i think changing all of that is a bigger task. i am merely trying to make this new feature idea map better to existing ways of defining commands.

Reply:
so yeah, i'm going to keep it for now since it's the same as everything we have in `routes.js` (except of course we don't have an HTTP method, and we call it `params` instead of `payloadParams` (I am open to renaming `params` in this PR to `payloadParams` for greater consistency).

but I agree with you, migrating our overall command definition to use json schema might be a good thing!